docs/dev_guide/code_standards.rst

```diff
@@ -1,33 +1,64 @@
+
+Language Standard
+-----------------
+
+English is the default language for all documentation strings and inline commands.
+Variables names should also be based on English words.
+In addition, the standard for spelling is American English.
+
+-  Packages must be compatible with Python 2.7, and 3.x (for 3.x compatibility, the six package should be used for new code).
+-  The code *should* be importable with no dependencies other than the Python Standard Library, NumPy, SciPy, matplotlib, and
+   components already required by SunPy.
+   Adding dependencies to SunPy will be considered but are highly discouraged.
+-  Documentation strings must be present for all public classes/methods/functions, and must follow the form outlined in the “Documentation Guidelines” document [tbc].
+   Additionally, examples or tutorials in the package documentation are strongly recommended.
+-  Unit tests are required for all public methods and functions, and should adhere to the standards set in the “Testing Guidelines”
+   document [tbc].
+-  C extensions will be allowed only when they provide a significant performance enhancement over pure python.
+   When C extensions are used, the python interface must meet interface guidelines, and the use of Cython is strongly recommended.
+-  If an external C library is needed, the source code for the library should be bundled with the SunPy core.
+   Additionally, the package must be compatible with using a system-installed library.
+-  Packages can include data in [path tbd] as long as it is less than about 100 kb.
+   If the data exceeds this size, it should be hosted outside the source code repository and downloaded using the approved
+   SunPy mechanism [tbc].
+-  All persistent configuration should be stored using the functions in
+   sunpy.config.
+-  General utilities necessary for but not specific to the package should be placed in the .utils module.
+   These utilities will be moved to the sunpy.utils module when the package is integrated into the core package. If a utility is already present in sunpy.utils, the package should always use that utility instead of re-implementing it in .utils.
+   If changes or enhancements are required of that utility than a separate pull request should be presented to the community
+-  Packages implementing many classes/functions not relevant to the component requested will not be accepted - the package should only
+   include the required functionality and relevant extensions.
+-  The use of short cryptic variable names is highly discouraged!
+-  All code should follow The Style Guide for Python (`PEP 8 <http://www.python.org/dev/peps/pep-0008/>`_)
+-  All code should follow the `coding style and conventions proposed by Astropy <http://docs.astropy.org/en/stable/development/codeguide.html#coding-style-conventions>`_.
+
+PEP8 Rules
+----------
+
+Additionally, all code that goes in the project should be checked using `PyLint <http://www.logilab.org/card/pylint_manual>`_.
```

Comment:
Do we want to use Pylint or recommend something else?

Reply:
flake8 is more focused on pep8 rather than general static code analysis. but 🤷‍